app/components/mixed_cards_component.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class MixedCardsComponent < ViewComponent::Base
```

Comment:
Would it be better to just have one array which is `cards` which contains objects like:
```rb
{
  type: :file,
  details: { ... }
}
```

So you can then iterate over them all like:
```
<% @cards.each do |card| %>
  <% case card_group[:type] %>
    <% when :file %>
      <%= render FileCardComponent.new(card[:details], title: @title, tracking: @tracking) %>
    <% when :external_link %>
      <%= render ExternalLinkCardComponent.new(card[:details]) %>
    <% end %>
  <% end %>
<% end %>
```

Meaning that we're not restricted by any ordering so if we so desired the external_link could be in the middle of the files

Reply:
Yeah its probably clearer that way. Will make the change